server/src/main/java/org/elasticsearch/indices/IndexingMemoryController.java

```diff
@@ -83,6 +83,16 @@
         Property.NodeScope
     );
 
+    /**
+     * If the uncommitted translog files for a given shard reach this age, the shard will be flushed.
+     */
+    public static final Setting<TimeValue> SHARD_UNCOMMITTED_TRANSLOG_MAX_AGE_SETTING = Setting.timeSetting(
+        "indices.memory.shard_uncommitted_translog_max_age",
+        TimeValue.timeValueHours(1),
```

Comment:
I'm not sure about the default and min values for this new configuration parameter, any feedback is welcomed.